datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -1087,9 +1105,48 @@ pub fn build_join_schema(
         }
     };
 
+    let mut right_id_key_groups = add_offset_to_identifier_key_groups(
```

Comment:
This code I think could also be put as a method on KeyIdentiferGroups which would make it easier to find 

```rust
impl KeyIdentiferGroups {
  /// Compute the output dependncies of self JOIN other with the specified join type
  fn join(&self, other:&Self, join_type: JoinType) {..}
}